src/test/java/org/beanpod/switchboard/controller/EncoderControllerTest.java

```diff
@@ -83,60 +95,50 @@ final void testRetrieveAllEncoders() {
   // When a encoder is available in the DB
   @Test
   final void testRetrieveEncoder() {
-    when(encoderDao.findEncoder(EncoderFixture.SERIAL_NUMBER)).thenReturn(Optional.of(encoderDTO));
+    when(encoderDao.findEncoder(user, EncoderFixture.SERIAL_NUMBER))
+        .thenReturn(Optional.of(encoderDTO));
     when(maintainDeviceStatus.maintainStatusField(anyList())).thenReturn(listOfDevices);
 
     ResponseEntity<EncoderDto> actualEncoder =
         encoderController.retrieveEncoder(EncoderFixture.SERIAL_NUMBER);
 
     assertNotNull(actualEncoder);
     assertEquals(200, actualEncoder.getStatusCodeValue());
-    assertEquals(encoder.getSerialNumber(), actualEncoder.getBody().getSerialNumber());
+    assertEquals(
+        encoder.getSerialNumber(),
+        Objects.requireNonNull(actualEncoder.getBody()).getSerialNumber());
```

Comment:
Why do we need to add `Objects.requireNonNull`? If it's because of a warning/code smell that the body might be null, it _should_ go away if you `assertNotNull(actualEncoder.getBody())`